packages/react/src/component.tsx

```diff
@@ -181,18 +210,24 @@ export function UploadDropzone<TRouter extends FileRouter>(
     setFiles(acceptedFiles);
   }, []);
 
-  const { startUpload, isUploading, permittedFileInfo } = useUploadThing(
-    $props.endpoint,
-    {
+  const { startUpload, isUploading, permittedFileInfo, uploadProgress } =
+    useUploadThing($props.endpoint, {
       onClientUploadComplete: (res) => {
         setFiles([]);
         if ($props.onClientUploadComplete) {
           $props.onClientUploadComplete(res);
         }
       },
       onUploadError: $props.onUploadError,
-    },
-  );
+    });
+  const progress: number = useMemo(() => {
```

Comment:
This is gonna trigger a _shitload_ of re-renders, minimum 10 per item being uploaded. Probably best to have an optional callback for tracking? Maybe a specific hook w/ progress? 🤔